lib/data.ts

```diff
@@ -0,0 +1,110 @@
+import sqlite3 from 'sqlite3'
+import { open } from 'sqlite'
+import { NO_DATA } from './constants'
+
+const BEGIN_TRANSACTION = 'BEGIN'
+const COMMIT_TRANSACTION = 'COMMIT'
+const QUERY_MAINTAINER_SQL = `
+  SELECT name, headline, bio, data as image
+  FROM Maintainers
+  LEFT JOIN Images ON Maintainers.image_id = Images.id
+  WHERE Maintainers.id = 1`
+const QUERY_SKILLS_SQL = `
+  SELECT name, rank
+  FROM Skills
+  ORDER BY rank DESC, name`
+const QUERY_ACHIEVEMENTS_SQL = `
+  SELECT title, subtitle, startDate, endDate, data as image
+  FROM Achievements
+  LEFT JOIN Images ON Achievements.image_id = Images.id
+  ORDER BY startDate DESC, title`
+const QUERY_MIN_ACHIEVEMENT_DATE_SQL = `
+  SELECT MIN(startDate) as min
+  FROM Achievements
+  LIMIT 1`
+const QUERY_HIGHLIGHTS_SQL = `
+  SELECT name, description, date, data as image
+  FROM Highlights
+  LEFT JOIN Images ON Highlights.image_id = Images.id
+  ORDER BY date DESC, name`
+const QUERY_CONTACT_INFORMATION_SQL = `
+  SELECT *
+  FROM ContactInformation`
+
+let dataCache: any = null;
+
+async function getData(): Promise<DataProps> {
+  try {
+    if (dataCache === null || process.env.ENVIRONMENT === 'DEV') {
+      const db = await open({
+        filename: process.env.SQLITE_DATABASE_PATH,
+        driver: sqlite3.Database
+      })
+    
+      await db.exec(BEGIN_TRANSACTION)
```

Comment:
Does this have to be transaction btw? Could you just

```
const QUERIES = [QUERY_MAINTAINER_SQL, QUERY_SKILLS_SQL, ...]

const [maintainer, skills, ...rest] = await Promise.all(
  QUERIES.map(db.all)
)
```

Reply:
No, not necessarily, I thought that the transaction would be the more performant option here. Although, I could run all the contents of the transaction in `Promise.all()`